text/0000-uefi-http-boot.md

```diff
@@ -0,0 +1,75 @@
+- Feature Name: UEFI HTTP Boot
+- Start Date: 2016-09-12
+- RFC PR:
+- Issue:
+
+# Summary
+[summary]: #summary
+
+UEFI supports new kind of boot via HTTP instead of traditional (and slow) TFTP. Implementing this in Foreman is possible with relatively small amount of changes.
+
+# Motivation
+[motivation]: #motivation
+
+Todays firmware is complicated and implementing a HTTP stack is possible, therefore we will see less TFTP in the future. Foreman already supports iPXE HTTP booting which is used for booting virtual machines with iPXE firmware built-in and UEFI HTTP Boot feature will be essentially the same. The feature can enable both UEFI HTTP and iPXE HTTP booting as the implementation requires only two main steps described below.
+
+# Detailed design
+[design]: #detailed-design
+
+In order to support HTTP booting, we need to have an HTTP service deployed. Until now, users were deploying their own services becaue foreman-proxy only provides TFTP service and HTTP API. As part of this RFC, I propose to implement a simple HTTP/HTTPS service as a foreman-proxy plugin serving all files from TFTP directory. Only regular files are supported, listing of directories must be disabled.
```

Comment:
> Can you imagine a real world scenario where https from a self-signed ca would work? Does UEFI firmware check https certificates?

Looks like there is no HTTPS support in the spec, this confirms it: https://firmwaresecurity.com/2015/05/09/new-uefi-http-boot-support-in-uefi-2-5/

But on the other hand iPXE does support HTTP UEFI chainboot, and then you can use https from there on. There is a patch pending from Red Hat engineers into iPXE project to allow embedding your own CA into the iPXE script itself (today it only works you can embed the CA into the firmware itself). This makes it quite usable.

But HTTP -> HTTPS is not super secure way of doing things...

I have no idea how https would work